opencost-ui.yaml

```diff
@@ -0,0 +1,86 @@
+package:
+  name: opencost-ui
+  version: 1.114.0
+  epoch: 0
+  dependencies:
+    runtime:
+      - nginx
+  copyright:
+    - license: Apache-2.0
+
+environment:
+  contents:
+    packages:
+      - busybox
+      - nodejs
+      - npm
+
+pipeline:
+  - uses: git-checkout
+    with:
+      expected-commit: dcbd21f018fa991ddbab8a71e356b9d74f115c31
+      repository: https://github.com/opencost/opencost-ui
+      tag: v${{package.version}}
+
+  - name: build
+    runs: |
+      npm install
+      npx parcel build src/index.html
+      mkdir -p ${{targets.contextdir}}/opt/ui/dist
+      mkdir -p ${{targets.contextdir}}/var/www
+      mkdir -p ${{targets.contextdir}}/usr/local/bin
+      adduser -u 1001 -D nginx
+      mkdir -p ${{targets.contextdir}}/var/log/nginx
+      cp ./dist/* ${{targets.contextdir}}/opt/ui/dist
+      cp THIRD_PARTY_LICENSES.txt ${{targets.contextdir}}/THIRD_PARTY_LICENSES.txt
+      cp ./docker-entrypoint.sh ${{targets.contextdir}}/usr/local/bin/
+
+subpackages:
+  - name: ${{package.name}}-nginx-config
+    description: nginx config for ${{package.name}}
+    dependencies:
+      # runtime:
+      #   - nginx-mainline-config
+      replaces:
+        - nginx-mainline-config
+        - nginx-config
```

Comment:
`nginx-config` is enough

Reply:
```
 workspace dir: /var/folders/9k/89xrbkvj08j5_yj_d867s3nw0000gn/T/melange-workspace-4111996962
2025/02/13 16:30:36 INFO   guest dir: /var/folders/9k/89xrbkvj08j5_yj_d867s3nw0000gn/T/melange-guest-332058334
2025/02/13 16:30:36 ERRO failed to test package: unable to build guest: unable to generate image: installing apk packages: installing packages: installing opencost-ui-nginx-config (ver:1.114.0-r0 arch:aarch64): unable to install files for pkg opencost-ui-nginx-config: unable to install file over existing one, different contents: etc/nginx/nginx.conf
```
if I didn't add nginx-mainline-config its failing with the error above

Reply:
is this some sort of a bug in packages?

Reply:
@developer-guy we need to also provide `nginx-config` as well:

```
provides:
  - nginx-config=0.${{package.full-version}}
 ```
 
 Or adjust `nginx-config` to set the provider priority

Reply:
should I need to remove nginx-mainline-config here?

Reply:
Remove all, but `nginx-config`. Ideally, we would use `provider-priority`, but there are several configs floating around currently. The idea behind this is we want `nginx-config` to always resolve to the latest version of the config and to not pull this one so we set a lower version with 0 appended at the beginning